tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -65,6 +66,29 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, X_unlabeled: DataFrame = N
         logger.log(20, f'Operating System:   {platform.system()}')
         logger.log(20, f'Platform Machine:   {platform.machine()}')
         logger.log(20, f'Platform Version:   {platform.version()}')
+        try:
+            # TODO: Make this logic smarter, incorporate training data size and potentially models to train into the logic to define the recommended disk space.
+            #  For example, `best_quality` will require more disk space than `medium_quality`, and HPO would require additional disk space.
+            disk_stats = ResourceManager.get_disk_usage(path=self.path)
+            disk_free_gb = disk_stats.free / 1e9
+            disk_total_gb = disk_stats.total / 1e9
```

Comment:
I notice you created class methods for `get_disk_usage_free`, `get_disk_usage_used`, and `get_disk_usage_total` but just used `get_disk_usage` and extracted the `free` and `total` values. Should we consider deleting the `get_disk_usage_{free,used,total}` methods if we aren't going to use them, it's easy to extract the members individually, and will avoid multiple calls to `shutil.disk_usage`?

Reply:
good point, I think they aren't necessary, so I've removed them.